File: eprofiler-core/src/test/java/cz/encircled/eprofiler/test/classes/TestClass.java
package cz.encircled.eprofiler.test.classes;

/**
 * @author Vlad on 23-May-16.
 */
public class TestClass {

    static {
        testStatic();
    }

    public static void testStatic() {
        System.out.println("testStatic");
    }

    public void firstMethod() {
        System.out.println("qwe");
    }

    public String someMethod() {
        String ret = "returnVal";
        try {
            Thread.sleep(20);
            yetAnotherMethod();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        anotherMethod("");
        return ret;
    }

    public void anotherMethod(String any) {
        try {
            Thread.sleep(20);
            yetAnotherMethod();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    public void yetAnotherMethod() {
        try {
            Thread.sleep(20);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

}

File: eprofiler-core/src/test/java/cz/encircled/eprofiler/test/classes/TestClass.kt
package cz.encircled.eprofiler.test.classes

/**
 * @author Vlad on 23-May-16.
 */
class TestClass {

    fun firstMethod() {
        println("qwe")
    }

    fun someMethod(): String {
        val ret = "returnVal"
        try {
            Thread.sleep(1000)
            yetAnotherMethod()
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }

        anotherMethod()
        return ret
    }

    fun anotherMethod() {
        try {
            Thread.sleep(2000)
            yetAnotherMethod()
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }

    }

    fun yetAnotherMethod() {
        try {
            Thread.sleep(500)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }

    }

    companion object {

        init {
            testStatic()
        }

        private fun testStatic() {
            println("testStatic")
        }
    }

}
